SU-SWS/react_paragraphs
# D8CORE-3743 Use limited cookie to temporarily save form data

# READY FOR REVIEW

# Summary
- Use a 5 second cookie to temporarily store form data in case of form validation issues.
   - The cookie is unique to the specific field
   - The cookie is only valid on the same url path so it won't exist on the actual node page or simultaneous node forms.
   - 5 seconds should be enough to load the react javascript and then the cookie goes away.

# Need Review By (Date)
- 4/21 (unless we get it in this week, we can push this out as a hotfix)

# Urgency
- medium

# Steps to Test
1. checkout this branch & clear browser caches
2. create a publication and choose the citation type "Article Newspaper/Magazine"
3. enter "01" into the month or day field (be sure it has a zero as the preceding character)
4. now populate some paragraph components with some values.
5. save and see the validation produces an error on the form
6. verify your paragraphs you populated are still there and configured correctly.

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
